model/markers.go

```diff
@@ -0,0 +1,286 @@
+
+// MarkerType will be an enum once we figure out the full list
+type MarkerType string
+
+// Marker is defined by the Wasabee IITC plugin.
+type Marker struct {
+	ID         MarkerID   `json:"ID"`
+	PortalID   PortalID   `json:"portalId"`
+	Type       MarkerType `json:"type"`
+	AssignedTo GoogleID   `json:"assignedTo"` // deprecated, use Assignments from Task
+	Task
+}
+
+// TODO use the logic from insertZone to unify insertMarker and updateMarker
+
+// insertMarkers adds a marker to the database
+func (opID OperationID) insertMarker(m Marker) error {
+	if m.State == "" {
+		m.State = "pending"
+	}
+
+	if !m.Zone.Valid() || m.Zone == ZoneAll {
+		m.Zone = zonePrimary
+	}
+
+	_, err := db.Exec("REPLACE INTO task (ID, opID, comment, taskorder, state, zone, delta) VALUES (?, ?, ?, ?, ?, ?, ?)",
+		m.ID, opID, MakeNullString(m.Comment), m.Order, m.State, m.Zone, m.DeltaMinutes)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	_, err = db.Exec("REPLACE INTO marker (ID, opID, PortalID, type) VALUES (?, ?, ?, ?)", m.ID, opID, m.PortalID, m.Type)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	// until clients get updated
+	if m.AssignedTo != "" {
+		m.Assignments = append(m.Assignments, m.AssignedTo)
+	}
+
+	if len(m.Assignments) > 0 {
+		err = m.Assign(m.Assignments, nil)
+		if err != nil {
+			log.Error(err)
+			return err
+		}
+	}
+
+	if len(m.DependsOn) > 0 {
+		err = m.SetDepends(m.DependsOn, nil)
+		if err != nil {
+			log.Error(err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (opID OperationID) updateMarker(m Marker, tx *sql.Tx) error {
+	if m.State == "" {
+		m.State = "pending"
+	}
+
+	if !m.Zone.Valid() || m.Zone == ZoneAll {
+		m.Zone = zonePrimary
+	}
+
+	assignmentChanged := false
+	if m.AssignedTo != "" {
+		var count uint8
+		err := tx.QueryRow("SELECT COUNT(*) FROM assignments WHERE taskID = ? AND opID = ? AND gid = ?", m.ID, opID, m.AssignedTo).Scan(&count)
+		if err != nil {
+			log.Error(err)
+			return err
+		}
+		if count != 1 {
+			assignmentChanged = true
+		}
+	}
+
+	_, err := tx.Exec("REPLACE INTO task (ID, opID, comment, taskorder, state, zone, delta) VALUES (?, ?, ?, ?, ?, ?, ?)",
+		m.ID, opID, MakeNullString(m.Comment), m.Order, m.State, m.Zone, m.DeltaMinutes)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	_, err = tx.Exec("REPLACE INTO marker (ID, opID, PortalID, type) VALUES (?, ?, ?, ?)", m.ID, opID, m.PortalID, m.Type)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	if assignmentChanged {
+		// until clients get updated
+		if m.AssignedTo != "" {
+			m.Assignments = append(m.Assignments, m.AssignedTo)
+		}
+
+		err = m.Assign(m.Assignments, tx)
+		if err != nil {
+			log.Error(err)
+			return err
+		}
+		messaging.SendAssignment(messaging.GoogleID(m.AssignedTo), messaging.TaskID(m.ID), messaging.OperationID(m.opID), m.State)
+	}
+
+	if len(m.DependsOn) > 0 {
+		err = m.SetDepends(m.DependsOn, tx)
+		if err != nil {
+			log.Error(err)
+			return err
+		}
+	}
```

Comment:
this part is similar to `updateLink`, is `updateTask` relevant ?

Reply:
it will all be unified in task after we can remove assignedTo